test/contraction/test_contraction_interface.cpp

```diff
@@ -0,0 +1,93 @@
+
+#include "ck/library/tensor_operation_instance/gpu/contraction_bilinear.hpp"
+
+#include "ck/library/utility/device_memory.hpp"
+
+using Pass     = ck::tensor_operation::element_wise::PassThrough;
+using Bilinear = ck::tensor_operation::element_wise::Bilinear;
+
+using F32 = float;
+using F64 = double;
+
+template <typename DataTypeA,
+          typename DataTypeB,
+          typename DataTypeC,
+          typename DataTypeD,
+          int NumDim>
+class ContractionDeviceWrapper
+{
+
+    protected:
+    using DeviceOp = ck::tensor_operation::device::DeviceContractionMultipleD<NumDim,
+                                                                              NumDim,
+                                                                              NumDim,
+                                                                              DataTypeA,
+                                                                              DataTypeB,
+                                                                              ck::Tuple<DataTypeC>,
+                                                                              DataTypeD,
+                                                                              Pass,
+                                                                              Pass,
+                                                                              Bilinear>;
+
+    public:
+    bool IsSupported() const
+    {
+        std::vector<ck::index_t> dummy_dims(NumDim * 2, 4);
+        std::vector<ck::index_t> dummy_strides(NumDim * 2, 1);
+
+        bool supported     = false;
+        const auto op_ptrs = ck::tensor_operation::device::instance::DeviceOperationInstanceFactory<
+            DeviceOp>::GetInstances();
+
+        for(auto& op_ptr : op_ptrs)
+        {
+            auto argument_ptr =
+                op_ptr->MakeArgumentPointer(nullptr,
+                                            nullptr,
+                                            std::array<const void*, 1>{nullptr},
+                                            nullptr,
+                                            dummy_dims,
+                                            dummy_strides,
+                                            dummy_dims,
+                                            dummy_strides,
+                                            std::array<std::vector<ck::index_t>, 1>{dummy_dims},
+                                            std::array<std::vector<ck::index_t>, 1>{dummy_strides},
+                                            dummy_dims,
+                                            dummy_strides,
+                                            Pass{},
+                                            Pass{},
+                                            Bilinear{1.f, 1.f});
+
+            supported = supported || op_ptr->IsSupportedArgument(argument_ptr.get());
+        }
+        return supported;
+    }
+};
+
```

Comment:
There is actually much more than dimensions and data types. Please take a look here https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/develop/include/ck/tensor_operation/gpu/device/impl/device_contraction_multiple_d_xdl_cshuffle.hpp#L583-L674 Ideally you should cover all those checks.

Reply:
Added testsfor non static checks. Static_checks are verified during compilation time.

Reply:
It seems to me that apart form `GridwiseGemm::CheckValidity` checks, which I'd rather not require here, there are 6 conditions checked here: https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/develop/include/ck/tensor_operation/gpu/device/impl/device_contraction_multiple_d_xdl_cshuffle.hpp#L611-L671. And you've added just two. To be clear, you don't have to go over all instances as you do right now. Just define single one for the test-case purposes, which would violate given condition and check if `IsSupportedArgument` function will catch it.

Reply:
I added tests for these conditions.  ( arg.a_grid_desc_.....GetLength(I2) % ABlockTransferSrcScalarPerVector == 0 are guaranteed by static  asserts in threadwise_tensor_slice_transfer).

Reply:
All other conditions are guaranteed by multiple static_asserts.